### DISCUSSION TITLE
Use multi level paths with one controller but not trigger on static assets

### DISCUSSION THREAD

**Discussion by mattiLeBlanc** [TypedProject/ts-express-decorators#1578] [Q&A] [answered]
Hi,

I am learning to work with Ts.ED I am a bit stuck with the following concept.

Context: 
we are serving an app that based on the URL, will load a pre-made template from Google Cloud Storage.
So for example `/australia` will call the controller which will fetch the PathParams `/:template` (which would be 'australia') and then we load an html template for 'australia' and return that response to the client.

Problem:
The Australia template will also have success page: `/australia/thank-you` and for that I need to fetch the `thank-you` page.
However, I am having trouble fetching that second level path.

My controller looks like: 

```typescript
@Controller('/')
@UseBefore(ClientGlobalMiddleware)
export class TemplateController {
  constructor(
    private templateService: TemplateService,
  ) { }

  @Get('/:template/:successPage')
  @Get('/:template')
  @View("template.hbs")
  public async get(
    @PathParams('template') template: string,
    @PathParams('successPage') successPage: string,
  ): Promise<any> {
  const content = await this.templateService.getContent(template, successPage);
   ...
```
Just using the one `@Get` for `:template` works, but when I add a second get for `/:template/:successPage'` the normal Australia page would not render anymore because now the controller is also  responding to static files like `/statics/bundle.min.js`.

How do I differentiate between path I actually want to respond to and assets  that should just be served.
Actually, using the one PathParams will also cause issue with `/favicon.ico` which is being processed by the controller.

So I guess I want to add an ignore list to paths linked to this controller.
How would I approach this?

UPDATE: one of the issues causing the loading of static assets and favicon was that in the html template, those paths didn't start with a forward slash. So they started relatively from the url path. 

However I still have trouble capturing `/:template' and `/:template/:successPage` and handle those 2 request separately. 
The moment I add a `@Get('/:template/:successPage')`, it is greedy and overrules  the `@Get('/:template')`.

**Comment by Romakita** [comment, marked as answer]
Hello @mattiLeBlanc 

I just released a new version of Ts.ED that let you to change the statics/controllers order. https://tsed.io/docs/serve-files.html#configuration

It should solve your issue by adding the `statics.hook = '$beforeRoutesInit'` options.

Also, you can simplify your Get decorator:

```typescript
@Controller('/')
@UseBefore(ClientGlobalMiddleware)
export class TemplateController {
  constructor(
    private templateService: TemplateService,
  ) { }

  @Get('/:template/:successPage?')
  @View("template.hbs")
  public async get(
    @PathParams('template') template: string,
    @PathParams('successPage') successPage: string,
  ): Promise<any> {
  const content = await this.templateService.getContent(template, successPage);
   ...
```

Replace all Get() by   @Get('/:template/:successPage?')

See you
Romain

**Reply by mattiLeBlanc** [reply]
Thanks @Romakita . I will see if it works for our setup. Will report back. Thanks for your assistence!

**Comment by mattiLeBlanc** [comment]
Thank you so much @Romakita . Works like a charm.
